plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseHiveConnectorTest.java

```diff
@@ -4978,13 +4977,49 @@ public void testUnsupportedSetPartitionColumnType()
     @Override
     protected void verifySetColumnTypeFailurePermissible(Throwable e)
     {
-        assertThat(e).hasMessageContaining("Cannot change type");
+        assertThat(e)
+                .hasMessageMatching(".*(Cannot change type|Unsupported storage format for changing column type).*");
+    }
+
+    @Override
+    protected List<SetColumnTypeSetup> setColumnTypeSetupData()
+    {
+        ImmutableList.Builder<SetColumnTypeSetup> setup = ImmutableList.builder();
+        for (HiveStorageFormat storageFormat : HiveStorageFormat.values()) {
+            if (storageFormat == REGEX) {
+                // REGEX format is read-only
+                continue;
+            }
+            setup.addAll(super.setColumnTypeSetupData().stream()
+                    .map(data -> data.withTableProperty("format = '%s'".formatted(storageFormat)))
```

Comment:
This is quite a few test cases.
For AVRO, RCBINARY, RCTEXT, SEQUENCEFILE, JSON, TEXTFILE, CSV, [REGEX] should be enough to have one test case eg changing from integer to bigint and checking message is "Unsupported storage format for changing column type"
(then you can simplify pattern in verifySetColumnTypeFailurePermissible)